### PR TITLE
Lower fcvt_from_{u,s}int for 8 and 16 bit ints

### PR DESCRIPTION
**Pull request by bjorn3**
Moved from https://github.com/cfallin/wasmtime/pull/13

### CODE REVIEW

File: cranelift/codegen/src/isa/aarch64/lower.rs [hunk 1]
```diff
@@ -2092,19 +2092,19 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(ctx: &mut C, insn: IRInst) {
             let out_bits = ty_bits(ctx.output_ty(insn, 0));
             let signed = op == Opcode::FcvtFromSint;
             let op = match (signed, in_bits, out_bits) {
-                (false, 32, 32) => IntToFpuOp::U32ToF32,
-                (true, 32, 32) => IntToFpuOp::I32ToF32,
-                (false, 32, 64) => IntToFpuOp::U32ToF64,
-                (true, 32, 64) => IntToFpuOp::I32ToF64,
+                (false, 8, 32) | (false, 16, 32) | (false, 32, 32) => IntToFpuOp::U32ToF32,
+                (true, 8, 32) | (true, 16, 32) | (true, 32, 32) => IntToFpuOp::I32ToF32,
+                (false, 8, 64) | (false, 16, 64) | (false, 32, 64)  => IntToFpuOp::U32ToF64,
```
[the remark this answers is not in the document]
Comment (bjorn3) [reply cranelift/codegen/src/isa/aarch64/lower.rs]: Oops, fixed.

**Comment by cfallin** [comment]
Floating-point tests pending in #1527; once that's merged, we can add tests here.

**Comment by cfallin** [comment]
#1527 has landed now so once we add tests to the vcode FP test added in that PR, this should be ready to land.

**Comment by bjorn3** [comment]
Rebased and added tests.

**Pull request merged**